snapshot/storage/bolt.go

```diff
@@ -372,7 +298,65 @@ func (ms *boltMetastore) Commit(ctx context.Context, key, name string) (id strin
 	return
 }
 
-func getSnapshot(bkt *bolt.Bucket, key string, ss *Snapshot) error {
+func withBucket(ctx context.Context, fn func(context.Context, *bolt.Bucket, *bolt.Bucket) error) error {
+	t, ok := ctx.Value(transactionKey{}).(*boltFileTransactor)
+	if !ok {
+		return errors.Errorf("no transaction in context")
```

Comment:
Updated, although not adding `IsNoTransaction` for now since it is not a valid program error. It should probably panic but doesn't seem necessary since the operation will be aborted already.